tally_ho/apps/tally/tests/views/test_super_admin.py

```diff
@@ -1476,3 +1479,100 @@ def test_edit_user_view_get(self):
         self.assertEquals(request.session['url_name'], 'user-list')
         self.assertEquals(request.session['url_param'], 'user')
         self.assertEquals(request.session['url_keyword'], 'role')
+
+    def test_view_result_forms_progress_by_form_state(self):
+        tally = issue_369_result_forms_data_setup(self.user)
+
+        view = views.FormProgressByFormStateView.as_view()
+        request = self.factory.get('/1/')
+        request.user = self.user
+        request.session = {}
+
+        response = view(request, tally_id=tally.pk)
+
+        # check that the response template is correct.
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.template_name,
+                         ['super_admin/form_progress_by_form_state.html'])
+        self.assertEqual(response.context_data['tally_id'], tally.pk)
+
+        response = response.render()
+        doc = BeautifulSoup(response.content, "xml")
+        ths = [th.text for th in doc.findAll('th')]
+        self.assertListEqual(
+            ths,
+            ['Race Type', 'Total forms', 'Unsubmitted', 'Intake',
+             'Data Entry 1', 'Data Entry 2', 'Corrections',
+             'Quality Control', 'Archived','Clearance', 'Audit']
+        )
+
+    def test_view_result_forms_progress_by_form_state_data_view(self):
+        tally = issue_369_result_forms_data_setup(self.user)
+
+        view = views.FormProgressByFormStateDataView.as_view()
+        request = self.factory.get('/')
+        request.user = self.user
+        request.session = {}
+
+        response = view(request, tally_id=tally.pk)
+
+        # check that the response template is correct.
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+        data = content["data"]
+        first_row_first_record = data[0]
```

Comment:
how about we distructure the array `data[0]` and assign meaningful variables to the items? So instead we will have:
```python
race_type, total_forms, unsubmitted_forms, ...etc = data[0]
```